src/coffea/nanoevents/methods/candidate.py

```diff
@@ -70,23 +71,29 @@ class PtEtaPhiECandidate(Candidate, vector.PtEtaPhiELorentzVector):
     pass
 
 
-awkward.behavior.update(
-    awkward._util.copy_behaviors(vector.LorentzVector, Candidate, behavior)
-)
-
-CandidateArray.ProjectionClass2D = vector.TwoVectorArray  # noqa: F821
-CandidateArray.ProjectionClass3D = vector.ThreeVectorArray  # noqa: F821
-CandidateArray.ProjectionClass4D = vector.LorentzVectorArray  # noqa: F821
-CandidateArray.MomentumClass = CandidateArray  # noqa: F821
+behavior.update(awkward._util.copy_behaviors(vector.LorentzVector, Candidate, behavior))
 
-PtEtaPhiMCandidateArray.ProjectionClass2D = vector.TwoVectorArray  # noqa: F821
-PtEtaPhiMCandidateArray.ProjectionClass3D = vector.ThreeVectorArray  # noqa: F821
-PtEtaPhiMCandidateArray.ProjectionClass4D = vector.LorentzVectorArray  # noqa: F821
-PtEtaPhiMCandidateArray.MomentumClass = PtEtaPhiMCandidateArray  # noqa: F821
+register_projection_classes(
+    CandidateArray,  # noqa: F821
+    vector.PolarTwoVectorArray,
+    vector.SphericalThreeVectorArray,
+    vector.LorentzVectorArray,
+    CandidateArray,  # noqa: F821
+)
+register_projection_classes(
+    PtEtaPhiMCandidateArray,  # noqa: F821
+    vector.PolarTwoVectorArray,
+    vector.SphericalThreeVectorArray,
+    vector.LorentzVectorArray,
+    PtEtaPhiMCandidateArray,  # noqa: F821
+)
+register_projection_classes(
```

Comment:
Ok this is my misunderstanding, I hadn't realized each case is somewhat different so the function doesn't in the end save any code duplication. Though at least for many of these the first 3 arguments are the same

Reply:
I think in this case what you had before was more readable code, is it ok to revert? No need to make separate PR

Reply:
arg, sorry, I thought convo was done, my bad

Reply:
I'll have to create a separate "revert" PR to fix this. There are some changes in this PR that should go in, like the extra tests and not modifying the global awkward behavior.

Reply:
Feel free to revert the subset of changes directly on the other PR and push, or make a revert PR if you prefer